src/ripple/app/main/Application.cpp

```diff
@@ -1618,9 +1618,45 @@ bool ApplicationImp::setup()
             RPC::ShardArchiveHandler::getStateDirectory(*config_)
             / "stateDatabase");
 
-        if(exists(stateDb) && is_regular_file(stateDb) &&
-           !RPC::ShardArchiveHandler::hasInstance())
-            RPC::ShardArchiveHandler::getInstance(*this);
+        try
+        {
+            if (exists(stateDb) &&
+                is_regular_file(stateDb) &&
+                !RPC::ShardArchiveHandler::hasInstance())
+            {
+                auto handler = RPC::ShardArchiveHandler::getInstance(*this);
+
+                if (!handler)
+                {
+                    JLOG(m_journal.fatal())
+                        << "Failed to create ShardArchiveHandler."
+                        << std::endl;
```

Comment:
It isn't the norm, nor necessary to add `std::endl` at the end of a log message.

Reply:
Done